controller/generic_controller.go

```diff
@@ -0,0 +1,28 @@
+package controller
+
+import (
+	"github.com/sirupsen/logrus"
+	"k8s.io/client-go/util/workqueue"
+)
+
+type genericController struct {
+	name   string
+	logger logrus.FieldLogger
+	queue  workqueue.RateLimitingInterface
```

Comment:
Why do you decide to include the queue here? I don't think this is aiming to be a controller framework, but a logger for the controllers.

Reply:
Clarified during discussion that the idea is not to build a controller framework.
This can be conceptually thought of an abstract base class, the reason to include the queue is to keep the name in sync.
I will rename the file to `base-controller` instead :)